### PR TITLE
GLTFLoader - Adding support for normalScale as part of the normalTexture

### PR DESCRIPTION
**Pull request by calrk**
The Unity > GLTF exporter allows the scales of the normals to be set on the normalTexture. 

I made a sample model with the left sphere at 0.5 scale, the center at 1.0 scale and the right at 2.0 scale.
[Spheres_20170823.zip](https://github.com/mrdoob/three.js/files/1243854/Spheres_20170823.zip)

The view in unity:
unity

However this property is not read by the current GLTFLoader
The result in gltf viewer - https://gltf-viewer.donmccurdy.com
threejs gltf

I'm adding this small change to read the 'scale' property of a normalTexture:
threejs gltf - fixed

Let me know what you think,
Clark

### CODE REVIEW

**Review by donmccurdy — CHANGES_REQUESTED** [review]
On [line 619](https://github.com/calrk/three.js/blob/d28ffdb5cb30cb8125f736dc710d9d270b350cbe/examples/js/loaders/GLTFLoader.js#L619), there's another reference to normalScale that I think needs to be changed to:

```js
material.normalScale = params.normalScale || new THREE.Vector2( 1, 1 );
```

... to have this fix also work on spec-gloss materials.

Thanks!

**Comment by calrk** [comment]
Sure, added that in now.

**Review by donmccurdy — APPROVED** [review]
Nice, thanks! 🙂

**Comment by mrdoob** [comment]
> ```js
> material.normalScale = params.normalScale || new THREE.Vector2( 1, 1 );
> ```

Hmm... `params.normalScale` can be a `THREE.Vector2`?

**Comment by donmccurdy** [comment]
If `params.normalScale` is defined, it will be coming from the `materialParams.normalScale = ...` line that @calrk just added. `materialParams` is passed into the `createMaterial()` method as `params`.

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -1613,6 +1613,11 @@ THREE.GLTFLoader = ( function () {
 
 				pending.push( parser.assignTexture( materialParams, 'normalMap', material.normalTexture.index ) );
 
+				if ( material.normalTexture.scale ) {
```
Comment (takahirox) [inline examples/js/loaders/GLTFLoader.js]: `material.normalTexture.scale` can't be zero?

Comment (donmccurdy) [inline examples/js/loaders/GLTFLoader.js]: Technically it could, sure. Let's do `material.normalTexture.scale !== undefined` then?

Comment (takahirox) [inline examples/js/loaders/GLTFLoader.js]: Yeah, I think it should be.

**Comment by calrk** [comment]
Ok, added in that check now too.

**Comment by mrdoob** [comment]
Hmm, maybe the logic could be simplified?

```js
material.normalScale = params.normalScale;
```

And then...

```js
materialParams.normalScale = new THREE.Vector2( 1, 1 );

if ( material.normalTexture.scale !== undefined ) {
	materialParams.normalScale.set( material.normalTexture.scale, material.normalTexture.scale );
}
```

**Comment by calrk** [comment]
I changed the structure of the default setting to be consistent with the other properties.

Did we want to keep this part in case this function is used in future elsewhere, so people don't have to define params.normalScale?
````
material.normalScale = params.normalScale === undefined ? new THREE.Vector2( 1, 1 ) : params.normalScale;
````

I'm not sure what happens if it was set to `null` or `undefined`, so just to be sure it gets the correct default?

File: examples/js/loaders/GLTFLoader.js [hunk 2]
```diff
@@ -616,7 +616,7 @@ THREE.GLTFLoader = ( function () {
 				material.bumpScale = 1;
 
 				material.normalMap = params.normalMap === undefined ? null : params.normalMap;
-				material.normalScale = new THREE.Vector2( 1, 1 );
+				material.normalScale = params.normalScale === undefined ? new THREE.Vector2( 1, 1 ) : params.normalScale;
```
Comment (mrdoob) [inline examples/js/loaders/GLTFLoader.js]: How about...

```js
if ( params.normalScale ) material.normalScale = params.normalScale;
```

Comment (calrk) [inline examples/js/loaders/GLTFLoader.js]: Sure, looks good. Changing it now.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**